splice_ck/src/main/java/com/splicemachine/ck/HBaseInspector.java

```diff
@@ -54,41 +57,66 @@ public HBaseInspector(final Configuration config) {
         this.config = config;
     }
 
-    public String scanRow(final String region, final String rowKey, final Utils.SQLType[] cols) throws Exception {
+    String limitString(int limit)
+    {
+        return "   .\n   .\n   .\n" + Utils.Colored.red(
+                "--- Result list was limited to " + limit + " entries, " + "result was cut off ---");
+    }
+
+    public String scanRow(final String region, final String rowKey, final Utils.SQLType[] cols,
+                          int limit, int versions, boolean hbase) throws Exception {
+
         StringBuilder result = new StringBuilder();
-        try(final ConnectionWrapper connectionWrapper = new ConnectionWrapper();
-            final ResultScanner rs = connectionWrapper.withConfiguration(config).connect().withRegion(region).scanSingleRowAllVersions(rowKey)) {
-            TableRowPrinter rowVisitor = new TableRowPrinter(cols == null ? null : new UserDefinedDataDecoder(cols, 4));
+        UserDataDecoder decoder = cols == null ? null : new UserDefinedDataDecoder(cols, 4);
+
+        // scan for one more than the limit so now if we truncated
+        int scanLimit = limit == 0 ? 0 : limit+1;
+        int count = 0;
+        ConnectionWrapper c = getCachedConnection().withRegion(region);
+        try(final ResultScanner rs = rowKey == null ?
+                c.scanVersions(scanLimit, versions) :
+                c.scanSingleRow(rowKey, versions)) {
             for(Result row : rs) {
+                if( count++ == limit ) {
+                    result.append(limitString(limit));
+                    break;
+                }
+                TableRowPrinter rowVisitor = new TableRowPrinter(decoder, hbase);
+                String hbaseStr = !hbase ? "" : " / " + com.splicemachine.primitives.Bytes.toStringBinary(row.getRow());
+                result.append(Utils.Colored.red("[ Row " + Hex.encodeHexString(row.getRow()) + hbaseStr + " ]\n"));
```

Comment:
please put in a separate method (something like: `generateRowId`) and put it in `Utils.java`, or, even better, make it part of the `processRow` method.

Reply:
done